src/algebra/algebra/operations.lean

```diff
@@ -186,6 +186,27 @@ end
 
 end decidable_eq
 
+lemma submodule.mul_eq_span_mul_set {R A : Type*} [comm_semiring R] [semiring A] [algebra R A]
+  (s t : submodule R A) : s * t = span R ((s : set A) * (t : set A)) :=
+by rw [← span_mul_span, span_eq, span_eq]
+
+lemma submodule.supr_mul_right {ι R A : Type*} [comm_semiring R] [semiring A] [algebra R A]
+  (s : ι → submodule R A) (t : submodule R A) : (⨆ i, s i) * t = ⨆ i, s i * t :=
+begin
+  conv_rhs { simp only [submodule.mul_eq_span_mul_set], },
+  rw [← (submodule.gi R A).gc.l_supr, set.supr_eq_Union, ← set.Union_mul_right, ← set.supr_eq_Union,
+    ← span_eq (span R ((⨆ (i : ι), ↑(s i)) * ↑t)), ← submodule.span_mul_span, span_eq, span_eq,
+    ← (submodule.gi R A).l_supr_u],
+end
+
+lemma submodule.supr_mul_left {ι R A : Type*} [comm_semiring R] [semiring A] [algebra R A]
+  (s : ι → submodule R A) (t : submodule R A) : t * (⨆ i, s i) = ⨆ i, t * s i :=
+begin
+  conv_rhs { simp only [submodule.mul_eq_span_mul_set], },
+  rw [← (submodule.gi R A).gc.l_supr, set.supr_eq_Union, ← set.Union_mul_left, ← set.supr_eq_Union,
```

Comment:
What's the statement of `(submodule.gi R A).gc.l_supr`? I feel like we probably have or should have a lemma for that.

Reply:
Yes its basically `submodule.span_Union` with a Union instead of supr. So pretty much there already, I just chose this way to stay in gc/supr land as long as possible though.